### ISSUE TITLE
[Deeplab] Label index does not match with code for dataset ADE20k?

### ISSUE THREAD

**Issue by haichaoyu** [tensorflow/models#4067]
Hello,

I tried to fine-tune Deeplab v3+ on ADE20k. I just found that in ADE20k, ignore_label is set to 0 while in the code, the ignore label is set to 255. Directly setting ignore_label to 0 in code might not work because the label index counts from 0.

Just want to confirm that this "problem" exists. Please correct me if I a wrong.

Thanks.

**Comment by walkerlala** [comment]
@haichaoyu Can you provide any reference for that (ie, ignore_label=0)? I would like to have further investigation.

And, @aquariusjay  is it possible for me to subscribe to issues which are tagged with [deeplab] on GitHub? That will enable to keep track on interesting deeplab-related issues.

**Comment by haichaoyu** [comment]
@walkerlala 
Hello, 

Sorry for my late reply. Please refer to [Ade20k dev kit](https://github.com/CSAILVision/sceneparsing) and search 'ignore'. Also we can also find some hints in the evaluation code [here](https://github.com/CSAILVision/sceneparsing/blob/master/evaluationCode/pixelAccuracy.m). I also checked the raw annotation images. The intensities range from 0 to 150.

Haichao

**Comment by bleedingfight** [comment]
@haichaoyu Thanks for your share,But I don't understand your said,I thinks ingore_label=255 stand dor the mask image's pixle is 255 the pixel will not be deal with,For my simple task  have object and background,I set object label in mask is 1,background label is 0,if I set  ingore_label=255 it will not effect my task.Can you tell me is my opinion right or not?

in your example,I think the labels is 0-150,0 stand for background,so I don't understand your code.Can you tell me more detail about ignore_label?thanks for your help.

**Comment by bleedingfight** [comment]
@haichaoyu thanks for your help,In your sence,you set 0-151 as your mask's label,0 stand for others,I your sence,you didn't need others,so you set it ignore,In your deeplabv3+,0 label will be ignore,so in your network 1-151label will be used.If I am understanding is right?but your classes is 150,even so your classes labels is 1-151,you will have 151 classes,why you have one another classes?

In my task I want use it in image to get lane's pixel,Should I set backgroud as others like your(olny one class) or I set it to as background to from a image get background and lane(two classes)?

**Comment by Eavis** [comment]
Hi @walkerlala,
ADE20K dataset description in https://github.com/tensorflow/models/blob/master/research/deeplab/datasets/segmentation_dataset.py
set ignore_label = 0
I think the dataset segmentation mask set background as 0 as well. 
I also see post suggest that ignore_label should be different from background label. Is that correct?
I am quite confused in this case. 

My task is to decrease the number of categories in the segmentation masks, so I only keep a few categories within 150 and set all others as background(0) Should I set ignore_label=0? 

Thanks!

**Comment by marlexct** [comment]
> Hi @walkerlala,
> ADE20K dataset description in https://github.com/tensorflow/models/blob/master/research/deeplab/datasets/segmentation_dataset.py
> set ignore_label = 0
> I think the dataset segmentation mask set background as 0 as well.
> I also see post suggest that ignore_label should be different from background label. Is that correct?
> I am quite confused in this case.
> 
> My task is to decrease the number of categories in the segmentation masks, so I only keep a few categories within 150 and set all others as background(0) Should I set ignore_label=0?
> 
> Thanks!

Hi, did you find an answer? I am trying to retrain Deeplab with ade20k, I did it with ignore_label = 0, but during training the loss just felt down oscillating between 0.2 and 0.3, then during evaluation I had bad results, miou very low =(, also vis.py brought some results, it wasn't the best.
